src/uwtools/templater.py

```diff
@@ -86,25 +85,19 @@ def set_template(argv):
         cfg.update(user_settings)
 
     # instantiate Jinja2 environment and template
-    env = Environment(
-        loader=FileSystemLoader(user_args.input_template),
-        undefined=StrictUndefined,
-    )
-    template = env.get_template('')
+    template = J2Template(cfg, user_args.input_template)
 
     # Gather the undefined template variables
-    j2_parsed = env.parse(env.loader.get_source(env, ''))
-    undeclared_variables = meta.find_undeclared_variables(j2_parsed)
+    undeclared_variables = template.undeclared_variables
```

Comment:
Does it make sense to move this line inside the `if user_args.values_needed` block? Maybe we don't have to spend the computational effort unless we need it.

The same principle holds for `rendered_template`.